StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1204StaticElementsMustAppearBeforeInstanceElements.cs

```diff
@@ -84,23 +84,29 @@ private static void HandleTypeDelcaration(SyntaxNodeAnalysisContext context)
 
         private static void HandleMemberList(SyntaxNodeAnalysisContext context, SyntaxList<MemberDeclarationSyntax> members)
         {
-            for (int i = 0; i < members.Count - 1; ++i)
+            var previousSyntaxKind = SyntaxKind.None;
+            var previousMemberStatic = true;
+            var previousMemberConst = true;
+            foreach (var member in members)
             {
-                var currentSyntaxKind = members[i].Kind();
+                var currentSyntaxKind = member.Kind();
                 currentSyntaxKind = currentSyntaxKind == SyntaxKind.EventFieldDeclaration ? SyntaxKind.EventDeclaration : currentSyntaxKind;
+                var modifiers = member.GetModifiers();
+                var currentMemberStatic = modifiers.Any(SyntaxKind.StaticKeyword);
```

Comment:
:x: The previous comment is not quite correct. If we don't distinguish between static and instance, an SA1204 warning would be reported when a constant follows an instance field, but only SA1203 should be reported in that case.